components/tools/OmeroJava/test/integration/ConfigurationServiceTest.java

```diff
@@ -203,4 +234,191 @@ public void testSupportedFormats() throws Exception {
             Assert.assertEquals(0, values.size());
         }
     }
+
+    /**
+     * Loads the enumeration.
+     * @return Object[][] data.
+     */
+    @DataProvider(name = "loadEnumData")
+    public Object[][] loadEnumData() throws Exception {
+        List<TestEnum> testParams = new ArrayList<TestEnum>();
+        Object[][] data = null;
+        ITypesPrx svc =  root.getSession().getTypesService();
+        List<IObject> objects = svc.getOriginalEnumerations();
+        List<String> types = svc.getEnumerationTypes();
+        Assert.assertTrue(types.size() > 0);
+        for (int i = 0; i < types.size(); i++) {
+            String type = types.get(i);
+            //tested via testSupportedFormats
+            if (type.equals("Format")) {
+                continue;
+            }
+            Set<String> original = new HashSet<String>();
+            type = "omero.model."+type+"I";
+            for (int j = 0; j < objects.size(); j++) {
+                IObject ho = objects.get(j);
+                if (ho.getClass().getName().equals(type)) {
+                    original.add(getEnumValue(ho));
+                }
+            }
+            testParams.add(new TestEnum(type, original));
+        }
+        int index = 0;
+        Iterator<TestEnum> j = testParams.iterator();
+        data = new Object[testParams.size()][1];
+        while (j.hasNext()) {
+            data[index][0] = j.next();
+            index++;
+        }
+        return data;
+    }
+
+    /**
+     * Returns the enumeration value.
+     *
+     * @param object The enumeration object.
+     * @return See above.
+     */
+    private String getEnumValue(IObject object)
+    {
+        RString value = null;
```

Comment:
For convenience another option might be some ugly trick with `PropertyUtils.getNestedProperty(object, "value.value")`.